sysfs/class_infiniband.go

```diff
@@ -121,16 +121,24 @@ func (fs FS) InfiniBandClass() (InfiniBandClass, error) {
 }
 
 // Parse one InfiniBand device.
+// Refer to https://www.kernel.org/doc/Documentation/ABI/stable/sysfs-class-infiniband
 func (fs FS) parseInfiniBandDevice(name string) (*InfiniBandDevice, error) {
 	path := fs.sys.Path(infinibandClassPath, name)
 	device := InfiniBandDevice{Name: name}
 
-	for _, f := range [...]string{"board_id", "fw_ver", "hca_type"} {
+	// fw_ver is exposed by all InfiniBand drivers since kernel version 4.10.
```

Comment:
We typically support older kernels than this. Maybe we should just make all of these files/values optional.

Reply:
Note that this PR is does not change the behaviour for `fw_ver`, which has been a mandatory requirement of this collector since it first landed in June 2019. It only makes the `board_id` optional, and `hca_type` was already optional.

I would be very surprised if anybody is running InfiniBand on kernels older than 4.10. I don't recall a single issue being logged about this collector failing due to missing `fw_ver`.